drivers/gpu/drm/vc4/vc4_crtc.c

```diff
@@ -728,10 +728,18 @@ static int vc4_crtc_atomic_check(struct drm_crtc *crtc,
 		if (conn_state->crtc != crtc)
 			continue;
 
-		vc4_state->margins.left = conn_state->tv.margins.left;
-		vc4_state->margins.right = conn_state->tv.margins.right;
-		vc4_state->margins.top = conn_state->tv.margins.top;
-		vc4_state->margins.bottom = conn_state->tv.margins.bottom;
+		if (memcmp(&conn_state->tv.margins, &vc4_state->margins,
+			   sizeof(vc4_state->margins))) {
+			vc4_state->margins.left = conn_state->tv.margins.left;
```

Comment:
Why don't you use `memcpy` for the update, since `memcmp` is used for comparison? If the margins structure were to include an extra field then you would potentially detect the need for an update but not actually change anything.

Reply:
And if you do, swap the `memcmp` parameters so they match the `memcpy`.

Reply:
Yup, happy to do that.
I only kept the old code to reduce the diff, but memcpy makes more sense.